lib/no_brainer/document/association/belongs_to.rb

```diff
@@ -30,12 +39,17 @@ def primary_key
       end
     end
 
-    def target_model
-      get_model_by_name(options[:class_name] || target_name.to_s.camelize)
+    def target_model(target_class = nil)
+      return if options[:polymorphic] && target_class.nil?
+
+      model_name = options[:class_name] || target_name.to_s.camelize
+      model_name = target_class if options[:polymorphic]
```

Comment:
I've updated the code as requested.

Regarding the `return if options[:polymorphic] && target_class.nil?` it is here in order to manage [the case where the polymorphic target is not loaded](https://github.com/NoBrainerORM/nobrainer/pull/275/files#diff-c8696e72aae0b75286348358ac501bf38266e9cd829092fba0421b8bcf32c76aR54):
1. on validating attributes from a document which have a `foreign_type` which is `nil`
2. `NoBrainer::Document::Association::BelongsTo`'s `polymorphic_read` method is fired and can't get the `target_class`
3. `return if options[:polymorphic] && target_class.nil?` prevents from having an error when calling `get_model_by_name` since the `model_name` would be `nil`.

Reply:
I'm still not convinced that it's semantically correct, but you probably know better than me as you have spend more time than me on this.